skia.lua

```diff
@@ -0,0 +1,1021 @@
+
+    dofile (_BUILD_DIR .. "/static_android_armv7_release.lua")
+
+    -- project specific configuration settings
+
+    configuration { "android_armv7_release" }
+
+      defines {
+        "SK_ARM_HAS_NEON",
+      }
+
+    -- -------------------------------------------------------------
+    -- configuration { "android_x86_debug" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_android_x86_debug.lua")
+
+    -- project specific configuration settings
+
+    configuration { "android_x86_debug" }
+
+      defines {
+        "SK_CPU_SSE_LEVEL=20",
+      }
+
+      files { opts_sse }
+
+    -- -------------------------------------------------------------
+    -- configuration { "android_x86_release" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_android_x86_release.lua")
+
+    -- project specific configuration settings
+
+    configuration { "android_x86_release" }
+
+      defines {
+        "SK_CPU_SSE_LEVEL=20",
+      }
+
+      files { opts_sse }
+
+    -- -------------------------------------------------------------
+    -- configuration { "android_arm64_debug" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_android_arm64_debug.lua")
+
+    -- project specific configuration settings
+
+    configuration { "android_arm64_debug" }
+
+      defines {
+        "SK_ARM_HAS_NEON",
+      }
+
+      files { opts_arm }
+      
+    -- -------------------------------------------------------------
+    -- configuration { "android_arm64_release" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_android_arm64_release.lua")
+
+    -- project specific configuration settings
+
+    configuration { "android_arm64_release" }
+
+      defines {
+        "SK_ARM_HAS_NEON",
+      }
+
+      files { opts_arm }
+      
+    -- -------------------------------------------------------------
+  end
+
+  if (_TARGET_IS_WINUWP) then
+    -- -------------------------------------------------------------
+    -- configuration { "winuwp" } == _TARGET_IS_WINUWP
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_winuwp.lua")
+
+    -- project specific configuration settings
+
+    configuration { "windows" }
+
+      defines {
+        "_CRT_SECURE_NO_WARNINGS",
+      }
+
+      includedirs {
+        "include/utils/win",
+        "src/utils/win",
+      }
+
+      buildoptions {
+        "/sdl",
+      }
+
+      -- Windows premake doesn't support per architecture files so add all opts and ifdef them in the code
+      files {
+        common_win,
+        opts_sse,
```

Comment:
For UWP, perhaps it is safer to just disable all optimizations. UWP has issues with neon still and ifdeffing the code may not be the best solution now that we use a fork.

Reply:
I think that comment (line 887) is old. We're not ifdefing the optimization files anymore. So the sse optimization files get built into the arm library, but the symbols will never get used because the code path to the optimized functions is ifdef'd out.